fem/fe_coll.cpp

```diff
@@ -2751,6 +2751,451 @@ ND_Trace_FECollection::ND_Trace_FECollection(const int p, const int dim,
 }
 
 
+ND_R1D_FECollection::ND_R1D_FECollection(const int p, const int dim,
+                                         const int cb_type, const int ob_type)
+   : FiniteElementCollection(p)
+{
+   MFEM_VERIFY(p >= 1, "ND_R1D_FECollection requires order >= 1.");
+   MFEM_VERIFY(dim == 1, "ND_R1D_FECollection requires dim == 1.");
+
+   if (cb_type == BasisType::GaussLobatto &&
+       ob_type == BasisType::GaussLegendre)
+   {
+      snprintf(nd_name, 32, "ND_R1D_%dD_P%d", dim, p);
+   }
+   else
+   {
+      snprintf(nd_name, 32, "ND_R1D@%c%c_%dD_P%d",
+               (int)BasisType::GetChar(cb_type),
+               (int)BasisType::GetChar(ob_type), dim, p);
+   }
+
+   for (int g = 0; g < Geometry::NumGeom; g++)
+   {
+      ND_Elements[g] = NULL;
+      ND_dof[g] = 0;
+   }
+
+   int op_type = BasisType::GetQuadrature1D(ob_type);
+   int cp_type = BasisType::GetQuadrature1D(cb_type);
+
+   // Error checking
+   if (Quadrature1D::CheckOpen(op_type) == Quadrature1D::Invalid)
+   {
+      const char *ob_name = BasisType::Name(ob_type);
+      MFEM_ABORT("Invalid open basis point type: " << ob_name);
+   }
+   if (Quadrature1D::CheckClosed(cp_type) == Quadrature1D::Invalid)
+   {
+      const char *cb_name = BasisType::Name(cb_type);
+      MFEM_ABORT("Invalid closed basis point type: " << cb_name);
+   }
+
+   ND_Elements[Geometry::POINT] = new ND_R1D_PointElement(p);
+   ND_dof[Geometry::POINT] = 2;
+
+   ND_Elements[Geometry::SEGMENT] = new ND_R1D_SegmentElement(p,
+                                                              cb_type,
+                                                              ob_type);
+   ND_dof[Geometry::SEGMENT] = 3 * p - 2;
+}
+
+const int *ND_R1D_FECollection::DofOrderForOrientation(Geometry::Type GeomType,
+                                                       int Or) const
+{
+   return NULL;
+}
+
+FiniteElementCollection *ND_R1D_FECollection::GetTraceCollection() const
+{
+   return NULL;
+}
+
+ND_R1D_FECollection::~ND_R1D_FECollection()
+{
+   for (int g = 0; g < Geometry::NumGeom; g++)
+   {
+      delete ND_Elements[g];
+   }
+}
+
```

Comment:
Extra line

Reply:
That's a matter of opinion, I suppose. This file often uses a double blank line between classes. See lines 1531, 1918, 1938, etc..